src/tests/ftest/util/server_utils.py

```diff
@@ -1052,3 +655,436 @@
+
+    def _check_information(self, key, section, retry=True):
+        """Check that the information dictionary is populated with data.
+
+        Args:
+            key (str): the information section to verify, e.g. "storage" or
+                "network"
+            section (str): The "response" key/section to verify
+            retry (bool): if set attempt to collect the server storage and
+                network information and call the method again. Defaults to True.
+
+        Raises:
+            ServerFailed: if output from the dmg command is missing or not in
+                the expected format
+
+        """
+        msg = ""
+        if key not in self.information:
+            msg = "Internal error: invalid information key: {}".format(key)
+        elif not self.information[key]:
+            self.log.info("Server storage/network information not collected")
+            if retry:
+                self.collect_information()
+                self._check_information(key, section, retry=False)
+        elif "status" not in self.information[key]:
+            msg = "Missing information status - verify dmg command output"
+        elif "response" not in self.information[key]:
+            msg = "No dmg {} scan 'response': status={}".format(
+                key, self.information[key]["status"])
+        elif section not in self.information[key]["response"]:
+            msg = "No '{}' entry found in information 'response': {}".format(
+                section, self.information[key]["response"])
+        if msg:
+            self.log.error(msg)
+            raise ServerFailed("ServerInformation: {}".format(msg))
+
+    def get_numa_node_info(self):
+        """Determine what NVMe and SCM storage is available per socket.
+
+        Raises:
+            ServerFailed: if output from the dmg storage scan is missing or
+                not in the expected format
+
+        Returns:
+            dict: [description]
+
+        """
+        self._check_information("storage", "HostStorage")
+        self._check_information("network", "HostFabrics")
+
+        data = {}
+        try:
+            _info = self.information["storage"]["response"]["HostStorage"]
+            for entry in _info.values():
+                hosts = entry["hosts"].split(":")[0]
+                data[hosts] = {}
+                if entry["storage"]["nvme_devices"]:
+                    data[hosts] = {"nvme": {}}
+                    for device in entry["storage"]["nvme_devices"]:
+                        data[hosts]["nvme"][device["pci_addr"]] = \
+                            device["socket_id"]
+                if entry["storage"]["scm_namespaces"]:
+                    data[hosts] = {"scm": {}}
+                    for device in entry["storage"]["scm_namespaces"]:
+                        data[hosts]["scm"][device["blockdev"]] = \
+                            device["numa_node"]
+        except KeyError as error:
+            raise ServerFailed(
```

Comment:
(pylint-raise-missing-from) Consider explicitly re-raising using the 'from' keyword